sign/impl.go

```diff
@@ -102,3 +105,18 @@ func (*defaultImpl) IsImageSignedInternal(
 
 	return len(signatures) > 0, nil
 }
+
+// TokenFromProviders will try the cosign OIDC providers to get an
+// oidc token from them.
+func (*defaultImpl) TokenFromProviders(ctx context.Context) (string, error) {
+	if !providers.Enabled(ctx) {
+		logrus.Warn("No OIDC provider enabled. Token cannot be obtained autmatically.")
```

Comment:
non-blocking nit, we can fix that later on: I'd prefer supporting to pass a logrus logger on Signer creation. Libraries which log out of the wild can be confusing. 

Yeah I know, we have them everywhere :grin:

Reply:
OK, done. It will reuse the logger at signer creation